traffic_ops/app/lib/API/DeliveryServiceRegexes.pm

```diff
@@ -222,6 +262,18 @@ sub delete {
 		return $self->forbidden();
 	}
 
+	my $ds = $self->db->resultset('Deliveryservice')->find( { id => $ds_id } );
+	if ( !defined($ds) ) {
+		#allow deletion if the ds is not valid
```

Comment:
I dont' understand this part. why not just return 404 not found if ds is invalid?